sharktank/sharktank/layers/base.py

```diff
@@ -121,15 +199,159 @@ def sample_inputs(
         """
         raise NotImplementedError()
 
+    def dynamic_shapes_for_export(
+        self,
+        batch_size: int | DynamicBatchSize | None = 1,
+        function: Optional[str] = None,
+    ) -> Dict[str, Any] | tuple[Any, ...] | list[Any] | None:
+        """During export the result is directly passed to the underlying export function."""
+        return None
+
+    def export_mlir(self, path: PathLike | None = None, /):
+        """Export the model into MLIR format.
+        Exporting is driven by the model's configuration.
+        Can be overridden in derived classes."""
+
+        if path is None:
+            path = self.config.mlir_path
+        if path is None:
+            raise ValueError("Missing MLIR export path.")
+
+        export_functions = [
+            ExportFunctionConfig(
+                function=self.default_export_function,
+                batch_sizes=self.default_export_batch_sizes,
+            )
+        ]
+        if self.config.export_functions is not None:
+            export_functions = self.config.export_functions
+
+        function_batch_size_pairs = {
+            export_function.function
+            or self.default_export_function: export_function.batch_sizes
+            or self.default_export_batch_sizes
+            for export_function in export_functions
+        }
+        from ..export import export_model_mlir
```

Comment:
Relating to my previous comment, https://github.com/nod-ai/shark-ai/blob/main/sharktank/sharktank/export.py is sort of orphaned outside of a subpackage at the moment. Is that intentional? Can we find it a better home too?